### PR TITLE
[7138] Bugfix/discovery server localhost

### PR DESCRIPTION
**Pull request by MiguelBarro**
Applies the same treatment given to the locators received via DATA, to the locators pass into the RemoteServerAttributes.
Basically the locators become localhost if possible to speed things up.

### CODE REVIEW

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1252/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1039/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1249/)

File: include/fastrtps/rtps/builtin/BuiltinProtocols.h [hunk 1]
```diff
@@ -71,6 +71,13 @@ class BuiltinProtocols
      */
     bool updateMetatrafficLocators(LocatorList_t& loclist);
 
+    /**
+     * Traverses the list of discover servers translating from remote to local locators
+     * if possible
+     * @param nf NetworkFactory used to make the translation
+     */
+    void transform_server_remote_locators(NetworkFactory & nf);
```
Comment (MiguelCompany) [inline include/fastrtps/rtps/builtin/BuiltinProtocols.h]:
```suggestion
    void transform_server_remote_locators(
            const NetworkFactory & nf);
```

File: src/cpp/rtps/builtin/BuiltinProtocols.cpp [hunk 2]
```diff
@@ -143,6 +145,21 @@ bool BuiltinProtocols::updateMetatrafficLocators(LocatorList_t& loclist)
     return true;
 }
 
+void BuiltinProtocols::transform_server_remote_locators(NetworkFactory & nf)
```
Comment (MiguelCompany) [inline src/cpp/rtps/builtin/BuiltinProtocols.cpp]:
```suggestion
void BuiltinProtocols::transform_server_remote_locators(
        const NetworkFactory & nf)
```

File: include/fastrtps/rtps/builtin/BuiltinProtocols.h (hunk 1 above)
Comment (MiguelBarro) [reply include/fastrtps/rtps/builtin/BuiltinProtocols.h]: done

File: src/cpp/rtps/builtin/BuiltinProtocols.cpp (hunk 2 above)
Comment (MiguelBarro) [reply src/cpp/rtps/builtin/BuiltinProtocols.cpp]: done

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1256/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1043/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1253/)

File: include/fastrtps/rtps/builtin/BuiltinProtocols.h [hunk 3]
```diff
@@ -69,7 +69,16 @@ class BuiltinProtocols
      * @param loclist LocatorList to update
      * @return True on success
      */
-    bool updateMetatrafficLocators(LocatorList_t& loclist);
+    bool updateMetatrafficLocators(
+        LocatorList_t& loclist);
```
Comment (MiguelCompany) [inline include/fastrtps/rtps/builtin/BuiltinProtocols.h]:
```suggestion
            LocatorList_t& loclist);
```

File: include/fastrtps/rtps/builtin/BuiltinProtocols.h [hunk 4]
```diff
@@ -69,7 +69,16 @@ class BuiltinProtocols
...
+
+    /**
+     * Traverses the list of discover servers translating from remote to local locators
+     * if possible
+     * @param nf NetworkFactory used to make the translation
+     */
+    void transform_server_remote_locators(
+        NetworkFactory & nf);
```
Comment (MiguelCompany) [inline include/fastrtps/rtps/builtin/BuiltinProtocols.h]:
```suggestion
            NetworkFactory & nf);
```

File: src/cpp/rtps/builtin/BuiltinProtocols.cpp [hunk 5]
```diff
@@ -137,12 +139,29 @@ bool BuiltinProtocols::initBuiltinProtocols(
     return true;
 }
 
-bool BuiltinProtocols::updateMetatrafficLocators(LocatorList_t& loclist)
+bool BuiltinProtocols::updateMetatrafficLocators(
+    LocatorList_t& loclist)
```
Comment (MiguelCompany) [inline src/cpp/rtps/builtin/BuiltinProtocols.cpp]:
```suggestion
        LocatorList_t& loclist)
```

File: src/cpp/rtps/builtin/BuiltinProtocols.cpp [hunk 6]
```diff
@@ -137,12 +139,29 @@ bool BuiltinProtocols::initBuiltinProtocols(
...
 {
     m_metatrafficUnicastLocatorList = loclist;
     return true;
 }
 
+void BuiltinProtocols::transform_server_remote_locators(
+    NetworkFactory & nf)
```
Comment (MiguelCompany) [inline src/cpp/rtps/builtin/BuiltinProtocols.cpp]:
```suggestion
        NetworkFactory & nf)
```

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1265/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1051/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1262/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1273/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1059/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1270/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1292/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1076/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1290/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1298/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1081/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1296/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1303/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1086/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1301/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1305/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1088/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1303/)

**Pull request merged**